apps/server/src/modules/files-storage/client/s3-client.adapter.ts

```diff
@@ -25,36 +25,47 @@ export class S3ClientAdapter implements IStorageClient {
 		this.logger.setContext(S3ClientAdapter.name);
 	}
 
-	async createBucket() {
+	// is public but only used internally
+	public async createBucket() {
 		try {
+			this.logger.debug({ action: 'create bucket', params: { bucket: this.config.bucket } });
+
 			const req = new CreateBucketCommand({ Bucket: this.config.bucket });
 			await this.client.send(req);
-		} catch (error) {
-			if (error instanceof Error) {
-				this.logger.error(`${error.message} "${this.config.bucket}"`);
+		} catch (err) {
+			if (err instanceof Error) {
+				this.logger.error(`${err.message} "${this.config.bucket}"`);
 			}
-			throw error;
+			throw new InternalServerErrorException(err, 'S3ClientAdapter:createBucket');
```

Comment:
I'm wondering if it is necessary and good to manually throw an `InternalServerErrorException` here. Don't we use the global exception filter of Nest (https://docs.nestjs.com/exception-filters)? Then there would be no reason to throw this. You would only manually throw errors that are not `InternalServerErrorException`s.

Reply:
let speak dirctly